plugins/cryptolib/cryptolib.cpp

```diff
@@ -0,0 +1,705 @@
+    {
+        doPKISign(__lenResult, __result, pDSM, lenInputdata, inputdata);
+    }
+    else
+    {
+        throw MakeStringException(-1, "Unable to create Digital Signature Manager");
+    }
+}
+
+CRYPTOLIB_API void CRYPTOLIB_CALL clPKISignBuff(size32_t & __lenResult, void * & __result,
+                                            const char * pkalgorithm,
+                                            const char * privatekeybuff,
+                                            const char * passphrase,
+                                            size32_t lenInputdata, const void * inputdata)
+{
+    verifyPKIAlgorithm(pkalgorithm);
+    IDigitalSignatureManager * pDSM = g_DSMCache.getSigningInstance(pkalgorithm, nullptr, privatekeybuff, passphrase);
+    if (pDSM)
+    {
+        doPKISign(__lenResult, __result, pDSM, lenInputdata, inputdata);
+    }
+    else
+    {
+        throw MakeStringException(-1, "Unable to create Digital Signature Manager");
+    }
+}
+
+CRYPTOLIB_API bool CRYPTOLIB_CALL clPKIVerifySignature(const char * pkalgorithm,
+                                                    const char * publickeyfile,
+                                                    const char * passphrase,
+                                                    size32_t lenSignature,const void * signature,
+                                                    size32_t lenSigneddata,const void * signeddata)
+{
+    verifyPKIAlgorithm(pkalgorithm);
+    IDigitalSignatureManager * pDSM = g_DSMCache.getVerifyingInstance(pkalgorithm, publickeyfile, nullptr, passphrase);
+    if (pDSM)
+    {
+        StringBuffer sbSig(lenSignature, (const char *)signature);
+        bool rc = pDSM->digiVerify(sbSig.str(), lenSigneddata, signeddata);
+        return rc;
+    }
+    else
+    {
+        throw MakeStringException(-1, "Unable to create Digital Signature Manager");
+    }
+}
+
+CRYPTOLIB_API bool CRYPTOLIB_CALL clPKIVerifySignatureBuff(const char * pkalgorithm,
+                                                       const char * publicKeyBuff,
+                                                       const char * passphrase,
+                                                       size32_t lenSignature, const void * signature,
+                                                       size32_t lenSigneddata, const void * signeddata)
+{
+    verifyPKIAlgorithm(pkalgorithm);
+    IDigitalSignatureManager * pDSM = g_DSMCache.getVerifyingInstance(pkalgorithm, nullptr, publicKeyBuff, passphrase);
+    if (pDSM)
+    {
+        StringBuffer sbSig(lenSignature, (const char *)signature);
+        bool rc = pDSM->digiVerify(sbSig.str(), lenSigneddata, signeddata);
+        return rc;
+    }
+    else
+    {
+        throw MakeStringException(-1, "Unable to create Digital Signature Manager");
+    }
+}
+
+
+//------------------------------------
+//Encryption helper
+//------------------------------------
+void doPKIEncrypt(size32_t & __lenResult,void * & __result,
```

Comment:
minor: safer to make non exported functions static - so they cannot clash with other functions.